### PR TITLE
Fixing the Helm docs

### PR DESCRIPTION
**Pull request by sekka1**
This fixes the helm-docs.

I am able to run it locally with the helm-docs version `v1.5.0` that matches the changes I am making here.  This should work in the GH Action pipeline now (the output of helm-docs is in markdown so the follow post of my local run here is a little funky =)).

```
 helm-docs  ./ci/helm-docs.sh                                                
INFO[2021-03-23T20:15:22-07:00] Found Chart directories [.]                  
INFO[2021-03-23T20:15:22-07:00] Generating README Documentation for chart /home/g44/Documents/locust.io/helm-chart/stable/locust 
# locust

![Version: 0.9.9](https://img.shields.io/badge/Version-0.9.9-informational?style=flat-square) ![AppVersion: 1.4.3](https://img.shields.io/badge/AppVersion-1.4.3-informational?style=flat-square)

A chart for locust

**Homepage:** <https://github.com/locustio/locust>

## How to install this chart

Add Locust chart repo:

```console
helm repo add locustio https://locustio.github.io/
```

A simple install with default values:

```console
helm install locustio/locust
```

To install the chart with the release name `my-release`:

```console
helm install my-release locustio/locust
```

To install with some set values:

```console
helm install my-release locustio/locust --set values_key1=value1 --set values_key2=value2
```

To install with custom values file:

```console
helm install my-release locustio/locust -f values.yaml
```

## Values

| Key | Type | Default | Description |
|-----|------|---------|-------------|
| affinity | object | `{}` |  |
| extraConfigMaps | object | `{}` | Any extra configmaps to mount for the master and worker. Can be used for extra python packages |
| extraLabels | object | `{}` |  |
| fullnameOverride | string | `""` |  |
| image.pullPolicy | string | `"IfNotPresent"` |  |
| image.repository | string | `"locustio/locust"` |  |
| image.tag | string | `"1.4.3"` |  |
| imagePullSecrets | list | `[]` |  |
| ingress.annotations | object | `{}` |  |
| ingress.enabled | bool | `false` |  |
| ingress.hosts[0].host | string | `"chart-example.local"` |  |
| ingress.tls | list | `[]` |  |
| loadtest.environment | object | `{}` | environment variables used in the load test |
| loadtest.locust_host | string | `"https://www.google.com"` | the host you will load test |
| loadtest.locust_lib_configmap | string | `""` | name of a configmap containing your lib |
| loadtest.locust_locustfile | string | `"main.py"` | the name of the locustfile |
| loadtest.locust_locustfile_configmap | string | `""` | name of a configmap containing your locustfile |
| loadtest.name | string | `"example"` | a name used for resources and settings in this load test |
| loadtest.pip_packages | list | `[]` | a list of extra python pip packages to install |
| master.args_include_default | bool | `true` | Whether to include default command args |
| master.command[0] | string | `"sh"` |  |
| master.command[1] | string | `"/config/docker-entrypoint.sh"` |  |
| master.image | string | `""` | A custom docker image including tag |
| master.logLevel | string | `"INFO"` | Log level. Can be INFO or DEBUG |
| master.pdb.enabled | bool | `false` | Whether to create a PodDisruptionBudget for the master pod |
| master.resources | object | `{}` | resources for the locust master |
| master.serviceAccountAnnotations | object | `{}` |  |
| master.strategy.type | string | `"RollingUpdate"` |  |
| nameOverride | string | `""` |  |
| nodeSelector | object | `{}` |  |
| podSecurityContext | object | `{}` |  |
| securityContext | object | `{}` |  |
| service.annotations | object | `{}` |  |
| service.extraLabels | object | `{}` |  |
| service.type | string | `"ClusterIP"` |  |
| tolerations | list | `[]` |  |
| worker.args_include_default | bool | `true` | Whether to include default command args |
| worker.command[0] | string | `"sh"` |  |
| worker.command[1] | string | `"/config/docker-entrypoint.sh"` |  |
| worker.hpa.enabled | bool | `false` |  |
| worker.hpa.maxReplicas | int | `100` |  |
| worker.hpa.minReplicas | int | `1` |  |
| worker.hpa.targetCPUUtilizationPercentage | int | `40` |  |
| worker.image | string | `""` | A custom docker image including tag |
| worker.logLevel | string | `"INFO"` | Log level. Can be INFO or DEBUG |
| worker.replicas | int | `1` |  |
| worker.resources | object | `{}` | resources for the locust worker |
| worker.serviceAccountAnnotations | object | `{}` |  |
| worker.strategy.type | string | `"RollingUpdate"` |  |

## Maintainers

| Name | Email | Url |
| ---- | ------ | --- |
| max-rocket-internet | max.williams@deliveryhero.com |  |
✅ No changes detected
```

### CODE REVIEW

**Comment by sekka1** [comment]
@max-rocket-internet Can you give this PR a review?  The helm-docs tests are passing now and it is now able to detect if there is a drift between the go template and the actual readme file.

File: ci/helm-docs.sh [hunk 1]
```diff
@@ -8,7 +8,7 @@ rm -f .helm-docs-md5sum-before .helm-docs-md5sum-after
 md5sum stable/*/README.md > .helm-docs-md5sum-before
 
 # Run helm-docs to generate all README.md files from the template
-helm-docs --template-file=../../ci/README.md.gotmpl
+helm-docs --dry-run --chart-search-root ./stable/locust
```
Comment (max-rocket-internet) [inline ci/helm-docs.sh:11]: How is dry run going to detect if there is changes in a PR that have not been reflected in the chart readme? It won't write anything?

**Comment by max-rocket-internet** [comment]
You also need to update the comment in the [main readme](https://github.com/locustio/helm-chart#running-ci-tests-locally) with this new version.

**Comment by max-rocket-internet** [comment]
If the helm-docs CI check was working in this PR right now then it would error because the chart README is not in sync with the chart, it shows `Version: 0.9.9` where it should be `Version: 0.9.10` 🙂

**Comment by sekka1** [comment]
Hmm...good point =).  Let me look into this.

**Comment by sekka1** [comment]
@max-rocket-internet ive push a change to the `helm-docs.sh` script to detect the changes.  The previous version the diff line wasnt producing any output so the `if` statement after that dropped into the good section.

After pushing this, the CI automatically ran this action: https://github.com/locustio/helm-chart/pull/7/checks?check_run_id=2205695092#step:4:12

Which failed.

Ive tested this all locally in the `jnorwood/helm-docs:v1.5.0` container and it working for me:

I get the initial failure like the above CI test:
```
/tmp/repo # ./ci/helm-docs.sh 
INFO[2021-03-26T22:34:19Z] Found Chart directories [.]                  
INFO[2021-03-26T22:34:19Z] Generating README Documentation for chart /tmp/repo/stable/locust 
Running diff:
--- .helm-docs-md5sum-before
+++ .helm-docs-md5sum-after
@@ -1 +1 @@
-5b9bda96ee7925ed5402228acb6a49d2  stable/locust/README.md
+cbe4026c8d4136b17cecc87e4caea496  stable/locust/README.md
🔴 These README.md files need to be updated with helm-docs: 

See main repo README.md for instructions
```

By running the `helm-docs.sh` script locally it changes my local `README.md` file:
```
git status -s                                                      
 M stable/locust/README.md
```
with these changes:
```diff
diff --git a/stable/locust/README.md b/stable/locust/README.md
index 727373f..4fc90b3 100644
--- a/stable/locust/README.md
+++ b/stable/locust/README.md
@@ -1,6 +1,6 @@
 # locust
 
-![Version: 0.9.9](https://img.shields.io/badge/Version-0.9.9-informational?style=flat-square) ![AppVersion: 1.4.3](https://img.shields.io/badge/AppVersion-1.4.3-informational?style=flat-square)
+![Version: 0.9.10](https://img.shields.io/badge/Version-0.9.10-informational?style=flat-square) ![AppVersion: 1.4.3](https://img.shields.io/badge/AppVersion-1.4.3-informational?style=flat-square)
 
 A chart for locust
```
So far so good and expected.  Now, with the badge updated, im going to run the `helm-doc.sh` again.  This time it should return a good status:
```
/tmp/repo # ./ci/helm-docs.sh 
INFO[2021-03-26T22:36:38Z] Found Chart directories [.]                  
INFO[2021-03-26T22:36:38Z] Generating README Documentation for chart /tmp/repo/stable/locust 
Running diff:
✅ No changes detected
/tmp/repo # echo $?
0
```
And it does.

Next im going to push the `README.md` changes here.  The CI test should pass now. :crossed_fingers:

**Comment by sekka1** [comment]
@max-rocket-internet The helm docs test passed here ^.  I think that tests out both cases in Action (failure and success).

File: ci/helm-docs.sh (hunk 1 above)
Comment (sekka1) [reply ci/helm-docs.sh]: yeah its not =)

**Comment by max-rocket-internet** [comment]
@sekka1 I don't really get the goal of this PR 🙂

> This fixes the helm-docs.

What was broken exactly?

**Comment by sekka1** [comment]
@max-rocket-internet sorry...should have made it more clear.  You mentioned here: https://github.com/locustio/helm-chart/pull/7#discussion_r600259627 it is in `--dry-run` mode and you are correct that this wouldnt do anything.

That was changed but there is more.

You also mentioned in a previous PR (Which i can't seem to find now...weird) that the version of the chart in the badge (https://github.com/locustio/helm-chart/tree/master/stable/locust) was not updated even though the helm-doc test passed.  You were correct again on this one.  So i looked into it.  The test passed even though there should have been a diff.  

If you run the helm-doc test (https://github.com/locustio/helm-chart/blob/master/.github/workflows/helm-docs.yml#L13) locally, change the helm chart version, and then run it on `master`, the test will pass.  Obviously it shouldnt pass b/c you just changed the chart version and the README.md will have a diff when helm-docs outputs it.  This turned out to be a bug in the diff-ing of the md5 outputs.  The diff command in `master` seems to always return nothing.  When going to the `if` statements after that, it will report a success.

What im trying to show here (https://github.com/locustio/helm-chart/pull/7#issuecomment-808564730) is a failure case and a success case to prove that this is working.

I could def do a better job of explaining.  Let me know if you need me to go into more details.

**Comment by max-rocket-internet** [comment]
Hey @sekka1 🙂

> it is in --dry-run mode and you are correct that this wouldnt do anything.
> That was changed but there is more.

But in master dry-run is not used. So this has not changed in this PR?

> You also mentioned in a previous PR that the version of the chart in the badge was not updated even though the helm-doc test passed.
>  The test passed even though there should have been a diff.

I don't think so. If there is any change in the chart at all, the version needs to be bumped and therefore helm-docs needs to run to generate the README, including the badge. I made an example PR https://github.com/locustio/helm-chart/pull/8 and you can see that helm-docs has failed as the README is now not up to date and chart-testing has failed as the chart version has not been bumped.

> If you run the helm-doc test locally

It's not supposed to be run locally, it's for github actions. In the main readme there is instructions for running the CI tests locally: https://github.com/locustio/helm-chart#running-ci-tests-locally


> If you run the helm-doc test (https://github.com/locustio/helm-chart/blob/master/.github/workflows/helm-docs.yml#L13) locally, change the helm chart version, and then run it on master, the test will pass.

Works for me, as README.md shows as changed?

```
$ git checkout master
Already on 'master'
Your branch is up to date with 'origin/master'.

$ sed -i 's/0.9.9/0.9.10/g' stable/locust/Chart.yaml

$ docker run --rm -v "$PWD:/helm-docs" jnorwood/helm-docs:v0.14.0 --template-file=../../ci/README.md.gotmpl
time="2021-03-31T07:45:43Z" level=info msg="Found Chart directories [stable/locust]"
time="2021-03-31T07:45:43Z" level=info msg="Generating README Documentation for chart stable/locust"

$ git status
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   stable/locust/Chart.yaml
	modified:   stable/locust/README.md

no changes added to commit (use "git add" and/or "git commit -a")
```

I'm not trying to be annoying, it's just hard to review a PR when the goal is not clear 😅

**Pull request closed without merging**